modules/openapi-generator/src/main/resources/go/model.mustache

```diff
@@ -35,9 +35,95 @@ type {{classname}} struct {
 {{#description}}
 	// {{{description}}}
 {{/description}}
-	{{name}} {{#isNullable}}*{{/isNullable}}{{{dataType}}} `json:"{{baseName}}{{^required}},omitempty{{/required}}"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
+	{{name}} *{{{dataType}}} `json:"{{baseName}},omitempty"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
+{{#isNullable}}    isExplicitNull{{name}} bool `json:"-"{{#withXml}} xml:"-"{{/withXml}}`{{/isNullable}}
 {{/vars}}
 }
 {{/isEnum}}
+
+{{^isEnum}}
+{{#vars}}
+// Get{{name}} returns the {{name}} field if non-nil, zero value otherwise.
+func (o *{{classname}}) Get{{name}}() {{dataType}} {
+	if o == nil || o.{{name}} == nil {
+		var ret {{dataType}}
+		return ret
+	}
+	return *o.{{name}}
+}
+
+// Get{{name}}Ok returns a tuple with the {{name}} field if it's non-nil, zero value otherwise
+// and a boolean to check if the value has been set.
+func (o *{{classname}}) Get{{name}}Ok() ({{dataType}}, bool) {
```

Comment:
the name doesn't really sonuds ok.. in favor of others e.g. `Get{{name}}IfExists`..

Reply:
I actually prefer `Get{{name}}Ok`, since it returns two values - the value of `{{name}}` and whether or not retrieving it was ok. With `IfExists`, I would expect the function to return only the value of `{{name}}` if it exists. YMMV.